L/licensecheck/bundled/main.go

```diff
@@ -1,27 +1,114 @@
 package main
 
+/*
+#include <stdlib.h>
+*/
 import "C"
 import (
 	"github.com/google/licensecheck"
 
+	"log"
+	"sync"
 	"unsafe"
 )
 
+var (
+	_licenses []licensecheck.License
+	_scanner  *licensecheck.Scanner = nil
+	_mu       sync.Mutex
+)
+
 //export License
 func License(msg *C.char) (**C.char, int, float64) {
+	_mu.Lock()
+	defer _mu.Unlock()
+
+	if len(_licenses) == 0 {
+		// _scanner do now work if there are no licenses, work-around such case
+		return (**C.char)(nil), 0, 0.0
+	}
```

Comment:
I think using a scanner can be done concurrently, we just need to make sure `_scanner` doesn't start pointing somewhere else mid-way through a scan. So we can do
```suggestion
    _mu.Lock()
    if len(_licenses) == 0 {
        _mu.Unlock()
        return (**C.char)(nil), 0, 0.0
    }
    scanner := _scanner            // copy pointer under lock
    _mu.Unlock()

    if scanner == nil {
        // shouldn't happen if you keep invariants, but be defensive
        return (**C.char)(nil), 0, 0.0
    }
```
and then use `scanner` instead of `_scanner` below, and we don't have to hold the lock the whole time.